rust/integration-tests/src/diff_tests/all_tests.rs

```diff
@@ -0,0 +1,181 @@
+                    .run(
+                        processor_config,
+                        test_type,
+                        move |conn: &mut PgConnection, txn_version: &str| {
+                            let mut json_data = match test_helper.load_data(conn, txn_version) {
+                                Ok(data) => data,
+                                Err(e) => {
+                                    eprintln!(
+                                        "[ERROR] Failed to load data for processor {} and transaction version {}: {}",
+                                        processor_name, txn_version, e
+                                    );
+                                    return Err(e);
+                                }
+                            };
+
+                            let expected_json_path = get_expected_json_path_fn(processor_name, txn_version);
+                            let expected_json = match read_and_parse_json(&expected_json_path) {
+                                Ok(json) => json,
+                                Err(e) => {
+                                    eprintln!(
+                                        "[ERROR] Error handling JSON for processor {} and transaction version {}: {}",
+                                        processor_name, txn_version, e
+                                    );
+                                    return Err(e);
+                                }
+                            };
+
+                            remove_inserted_at(&mut json_data);
+                            // Remove the Err handling around assert_json_eq!, as it doesn't return a Result type
+                            assert_json_eq!(&json_data, &expected_json);
+
+                            println!(
+                                "[INFO] Test passed for processor {} and transaction version: {}",
+                                processor_name, txn_version
+                            );
+                            Ok(())
+                        },
+                    )
+                    .await
+                    .unwrap();
+            }
+        }
+    }
+
+    // Helper function to read and parse JSON files
+    fn read_and_parse_json(path: &str) -> Result<serde_json::Value, anyhow::Error> {
+        match fs::read_to_string(path) {
+            Ok(content) => match serde_json::from_str::<serde_json::Value>(&content) {
+                Ok(json) => Ok(json),
+                Err(e) => {
+                    eprintln!("[ERROR] Failed to parse JSON at {}: {}", path, e);
+                    Err(anyhow::anyhow!("Failed to parse JSON: {}", e))
+                },
+            },
+            Err(e) => {
+                eprintln!("[ERROR] Failed to read file at {}: {}", path, e);
+                Err(anyhow::anyhow!("Failed to read file: {}", e))
+            },
+        }
+    }
+
+    fn get_processor_map() -> HashMap<String, Arc<Box<dyn ProcessorTestHelper>>> {
```

Comment:
this looks weird.. why `Arc` here?

Reply:
the closure in test_context.run requires a 'static lifetime, and the processor_map is a local variable so it doesnt have a 'static lifetime. so I used arc b/c the value itself doesn't need to be copied or cloned.. and I dont' have to implement clone for the trait and Box<dyn ProcessorTestHelper>..

Reply:
i feel if you make the map static, and use the reference in these places, it could work. 

not a blocking issue